python/src/iceberg/transforms.py

```diff
@@ -0,0 +1,520 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import base64
+import math
+import re
+import struct
+from datetime import datetime, timedelta
+from decimal import Decimal
+
+import mmh3  # type: ignore
+import pytz
+
+from .types import (
+    BinaryType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    LongType,
+    StringType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    Type,
+    UUIDType,
+)
+
+
+class Transform(object):
+    """
+    Transform base class for concrete transforms. The default implementation is for VoidTransform.
+    """
+
+    _EPOCH = datetime.utcfromtimestamp(0)
+
+    @staticmethod
+    def _human_day(day_ordinal):
+        time = Transform._EPOCH + timedelta(days=day_ordinal)
+        return "{0:0=4d}-{1:0=2d}-{2:0=2d}".format(time.year, time.month, time.day)
+
+    @staticmethod
+    def _unscale_decimal(decimal_value: Decimal):
+        value_tuple = decimal_value.as_tuple()
+        return int(
+            ("-" if value_tuple.sign else "")
+            + "".join([str(d) for d in value_tuple.digits])
+        )
+
+    def __init__(self, transform_string: str, repr_string: str):
+        self._transform_string = transform_string
+        self._repr_string = repr_string
+
+    def __repr__(self):
+        return self._repr_string
+
+    def __str__(self):
+        return self._transform_string
+
+    def apply(self, value):
+        return None
+
+    def can_transform(self, target: Type) -> bool:
+        return True
+
+    def get_result_type(self, source: Type) -> Type:
+        return source
+
+    def preserve_order(self) -> bool:
```

Comment:
Was this changed from "preserves" to "preserve" for some Python convention?

Reply:
So far, seems most of methods I have seen starting with a `verb` do not append a `s`. I am OK either with or without `s`. We should keep it consistent across all methods in the iceberg python library.

Reply:
What do you mean "most methods"? In Iceberg Java, we use the correct form for English: "the identity transform *preserves* order".

Reply:
I find that this is not always true and sometimes inconsistent. For example, `getter` or `setter` methods, `compare` versus `equals`, `find` vs `contains`, etc.
Also, I feel `most methods` in python seem not having `s`, e.g.  for regex, python `re.match(pattern, string)` versus Java `Pattern.matches(pattern, string)`. 
@rdblue  Is there any convention rule here for those names?

Reply:
If there is a published PEP that requires dropping the `s`, then we can go with that.

Otherwise, use the form that is correct when reading code as a sentence:

```
if transform.preserves_order():
    pass
```

The form with `s` is used here because when you read it you get "if the transform preserves order". Without the `s` is not grammatically correct in that sentence.

Reply:
👌  I updated the name of preserves_order and satisfies_order_of.